Google One Tap implementation

kviktor opened the pull request:
## Proposed changes

The changes adds Google One Tap as a backend

## Checklist


- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added documentation to https://github.com/python-social-auth/social-docs

## Other information

You can test the changes with this dummy project https://github.com/kviktor/django-onetap-example

I haven't yet added tests or documentation but if the changes look okay I'll add them too.

I have 2 questions tho I'm not exactly sure about, maybe someone can help out
- I've added google-auth as a dependency, technically it's just downloading Google's cert files then verifying the jwt token however I didn't want to reimplement the wheel for it and it's a bit safer this way imo
- as it requires a new dependency I've added a new Python module for it, is that ok? (I can move it into `google.py` with catching ImportError and raising an exception if the library is missing in the class)

Docs at https://github.com/python-social-auth/social-docs/pull/256

nijel commented on social_core/backends/google_onetap.py:
```diff
@@ -0,0 +1,53 @@
+        if not csrf_token_cookie:
+            # csrf_token_cookie can be missing due to
+            # https://issuetracker.google.com/issues/226157137
+            return
```
This makes the whole CSRF verification mostly useless as it does not verify requests without cookies.

kviktor replied:
I agree, I was thinking of making it configurable so users who have everything on a single domain have a sane default and users with who have the auth part on a different one can disable it, I'll add that with the tests.

nijel reviewed (commented):
Looks okay at first look. Can you please add tests and documentation?

kviktor commented:
@nijel I've added some tests, because it's not on oauth flow they are a bit different.

Also added docs at https://github.com/python-social-auth/social-docs/pull/256

Also switched from urllib3 to requests as I've just noticed this library already depends on it and that requirement doesn't need that weird packaging library workaround.
